src/gpu/intel/jit/ngen/ngen_pseudo.hpp

```diff
@@ -38,6 +38,18 @@ void min(const InstructionModifier &mod, const RegData &dst, const RegData &src0
     sel<DT>(mod | lt | f0[0], dst, src0, src1);
 }
 #endif
+
+template <typename DT = void>
+void amax_(const InstructionModifier &mod, const RegData &dst, const RegData &src0, const RegData &src1) {
+    sel<DT>(mod | ge | f0[0], dst, src0, abs(src1));
+}
+#ifndef NGEN_WINDOWS_COMPAT
+template <typename DT = void>
+void amax(const InstructionModifier &mod, const RegData &dst, const RegData &src0, const RegData &src1) {
+    sel<DT>(mod | ge | f0[0], dst, src0, abs(src1));
+}
+#endif
```

Comment:
I don't think there's any need to define an `amax` pseudo-instruction within nGEN, since it's not a widely-used operation and nGEN already provides the building blocks for it. Instead, you can keep the implementation completely within jit_reduction_injector.